internal/jobcontainers/jobcontainer.go

```diff
@@ -645,7 +763,43 @@ func systemProcessInformation() ([]*winapi.SYSTEM_PROCESS_INFORMATION, error) {
 // Takes a string and replaces any occurrences of CONTAINER_SANDBOX_MOUNT_POINT with where the containers' volume is mounted, as well as returning
 // if the string actually contained the environment variable.
 func (c *JobContainer) replaceWithMountPoint(str string) (string, bool) {
-	newStr := strings.ReplaceAll(str, "%"+sandboxMountPointEnvVar+"%", c.sandboxMount[:len(c.sandboxMount)-1])
-	newStr = strings.ReplaceAll(newStr, "$env:"+sandboxMountPointEnvVar, c.sandboxMount[:len(c.sandboxMount)-1])
+	mountPoint := c.rootfsLocation
+	newStr := strings.ReplaceAll(str, "%"+sandboxMountPointEnvVar+"%", mountPoint[:len(mountPoint)-1])
+	newStr = strings.ReplaceAll(newStr, "$env:"+sandboxMountPointEnvVar, mountPoint[:len(mountPoint)-1])
 	return newStr, str != newStr
 }
+
+func (c *JobContainer) bindSetup(ctx context.Context, s *specs.Spec) (err error) {
+	// Must be upgraded to a silo so we can get per silo bindings for the container.
+	if err := c.job.PromoteToSilo(); err != nil {
+		return err
+	}
+	// Union the container layers. Luckily bindflt works fine with \\?\volume paths, so we don't need to
+	// mount the volume anywhere friendly (for example, somewhere under C:\) first.
+	if err := c.mountLayers(ctx, c.id, s, ""); err != nil {
+		return fmt.Errorf("failed to mount container layers: %w", err)
+	}
+	rootfsLocation := defaultSiloRootfsLocation
+	if loc := customRootfsLocation(s.Annotations); loc != "" {
```

Comment:
Is there validation that the annotation specifies somewhere on c: or can it be anywhere?

Reply:
Should be fine on another drive